norfair/lib_metrics.py

```diff
@@ -0,0 +1,336 @@
+        self.matrixes_predictions = []
+        self.paths = []
+
+    def create_acc(self, input_path=None):
+        if not hasattr(self, 'matrixes_predictions'):
+            self.matrixes_predictions=[]
+        if not hasattr(self, 'paths'):
+            self.paths=[]
+
+
+        if input_path is None:
+            raise ValueError(
+                "You must set 'input_path' argument when creating new accumulator"
+            )
+
+        file_name = os.path.split(input_path)[1]
+
+        self.frame_number = 1
+        # save the path of this video in a list
+        self.paths = np.hstack((self.paths, input_path))
+        # initialize a matrix where we will save our predictions for this video (in the MOTChallenge format)
+        self.matrix_predictions = []
+
+        # initialize progress bar
+        seqinfo_path = os.path.join(input_path, "seqinfo.ini")
+        length = search_value_on_document(seqinfo_path, "seqLength")
+        self.progress_bar_iter = track(
+            range(length - 1), description=file_name, transient=False
+        )
+
+    def update(self, predictions=None):
+        # get the tracked boxes from this frame in an array
+        for obj in predictions:
+            new_row = [
+                self.frame_number,
+                obj.id,
+                obj.estimate[0, 0],
+                obj.estimate[0, 1],
+                obj.estimate[1, 0] - obj.estimate[0, 0],
+                obj.estimate[1, 1] - obj.estimate[0, 1],
+                -1,
+                -1,
+                -1,
+                -1,
+            ]
+            if np.shape(self.matrix_predictions)[0] == 0:
+                self.matrix_predictions = new_row
+            else:
+                self.matrix_predictions = np.vstack((self.matrix_predictions, new_row))
+        self.frame_number += 1
+        # advance in progress bar
+        try:
+            next(self.progress_bar_iter)
+        except StopIteration:
+            self.matrixes_predictions.append(self.matrix_predictions)
+            return
+
+    def compute_metrics(
+        self,
+        metrics=None,
+        generate_overall=True,
+    ):
+        if metrics == None:
+            metrics = list(mm.metrics.motchallenge_metrics)
+
+        self.summary = eval_motChallenge(
+            matrixes_predictions=self.matrixes_predictions,
+            paths=self.paths,
+            metrics=metrics,
+            generate_overall=generate_overall,
+        )
+
+        return self.summary
+
+    def save_metrics(self, save_path=".", file_name="metrics.txt"):
+
+        # create file to save metrics
+        if not os.path.exists(save_path):
+            os.makedirs(save_folder)
+        metrics_path = os.path.join(save_path, file_name)
+        metrics_file = open(metrics_path, "w+")
+        metrics_file.write(self.summary)
+        metrics_file.close()
+
+    def print_metrics(self):
+        print(self.summary)
+
+
+def load_motchallenge(matrix_data, min_confidence=-1):
```

Comment:
If this is a modification of a function in the MOT library, it would good to add a short text somewhere saying so, and explaining what parts where modified.

Reply:
Ok